pyuvdata/uvdata.py

```diff
@@ -833,6 +833,50 @@ def phase_to_time(self, time, phase_frame='icrs', use_ant_pos=False):
         self.phase(zenith_ra, zenith_dec, epoch='J2000', phase_frame=phase_frame,
                    use_ant_pos=use_ant_pos)
 
+    def set_uvws_from_antenna_positions(self, allow_phasing=False):
+        """
+        Calculate UVWs based on antenna_positions
+
+        Args:
+            allow_phasing: Option for phased data. If data is phased and
+            allow_phasing is set, data will be unphased, UVWs will be
+            calculated, and then data will be rephased. Script will error if
+            data is phased and allow_phasing is not set. Default is False
+        """
+        phase_type = self.phase_type
+        if phase_type == 'phased':
+            if allow_phasing:
+                warnings.warn('Warning: Data will be unphased and rephased '
+                              'to calculate UVWs.'
+                              )
+                phase_center_ra = self.phase_center_ra
+                phase_center_dec = self.phase_center_dec
+                phase_center_epoch = self.phase_center_epoch
+                self.unphase_to_drift()
+            else:
+                raise ValueError('UVW calculation requires unphased data. '
+                                 'Use unphase_to_drift or set '
+                                 'allow_phasing=True.'
+                                 )
+        antenna_locs_ENU = uvutils.ENU_from_ECEF(
+            (self.antenna_positions + self.telescope_location).T,
+            *self.telescope_location_lat_lon_alt
+        ).T
+        uvw_array = np.zeros((self.baseline_array.size, 3))
+        for baseline in list(set(self.baseline_array)):
```

Comment:
Note that the baselines are repeated often in the baseline array. You might want to only grab the unique ones, something like what we do in the `phase` function.

Reply:
"set" should be taking care of that

Reply:
yes, you're right. I missed that when I read it the first time.